Source/WebCore/editing/cocoa/EditorCocoa.mm

```diff
@@ -178,15 +179,17 @@
 void Editor::writeSelection(PasteboardWriterData& pasteboardWriterData)
 {
     Ref document = protectedDocument();
-    auto string = selectionAsAttributedString(document);
 
     PasteboardWriterData::WebContent webContent;
     webContent.contentOrigin = document->originIdentifierForPasteboard();
     webContent.canSmartCopyOrDelete = canSmartCopyOrDelete();
     webContent.dataInWebArchiveFormat = selectionInWebArchiveFormat();
-    webContent.dataInRTFDFormat = [string containsAttachments] ? dataInRTFDFormat(string.get()) : nullptr;
-    webContent.dataInRTFFormat = dataInRTFFormat(string.get());
-    webContent.dataInAttributedStringFormat = archivedDataForAttributedString(string.get());
+    if (document->settings().writeRichTextDataWhenCopyingOrDragging()) {
```

Comment:
this could probably be refactored into a generic function since it’s the same few lines as above, but that’s probably overkill.

Reply:
It's not _too_ unwieldy, though I'd need to pull the static `dataInRTFDFormat` / `dataInRTFFormat` helpers out into the `public` section so that I can use them here:

```
template<typename PasteboardContent>
void populateRichTextDataIfNeeded(PasteboardContent& content, const Document& document)
{
    if (!document.settings().writeRichTextDataWhenCopyingOrDragging())
        return;

    auto string = selectionAsAttributedString(document);
    content.dataInRTFDFormat = [string containsAttachments] ? Editor::dataInRTFDFormat(string.get()) : nullptr;
    content.dataInRTFFormat = Editor::dataInRTFFormat(string.get());
    content.dataInAttributedStringFormat = archivedDataForAttributedString(string.get());
}
```

Thanks for the review!